client/ayon_core/hosts/substancepainter/plugins/load/load_mesh.py

```diff
@@ -8,10 +9,120 @@
+
+    Args:
+        template_name (str): name of the template from the setting
+        project_templates (dict): project template data from the setting
+
+    Returns:
+        dict: data to be used as attributes for Substance Project Creation
+    """
+    attributes_data = {}
+    for template in project_templates:
+        if template["name"] == template_name:
+            attributes_data.update(template)
+    attributes_data["normal_map_format"] = _convert(
+        attributes_data["normal_map_format"])
+    attributes_data["project_workflow"] = _convert(
+        attributes_data["project_workflow"])
+    attributes_data["tangent_space_mode"] = _convert(
+        attributes_data["tangent_space_mode"])
+    attributes_data.pop("name")
+    attributes_data.pop("preserve_strokes")
+    return attributes_data
+
+
+def parse_subst_attrs_reloading_mesh(template_name, project_templates):
+    """Function to parse the substances attributes ('import_cameras'
+        and 'preserve_strokes') for reloading mesh
+        with the existing projects.
+
+    Args:
+        template_name (str): name of the template from the setting
+        project_templates (dict): project template data from the setting
+
+    Returns:
+        dict: data to be used as attributes for reloading mesh with the
+            existing project
+    """
+    attributes_data = {}
+    for template in project_templates:
+        if template["name"] == template_name:
+            for key, value in template.items():
+                if isinstance(value, bool):
+                    attributes_data.update({key: value})
+    return attributes_data
+
+
+class SubstanceProjectConfigurationWindow(QtWidgets.QDialog):
+    """The pop-up dialog allows users to choose material
+    duplicate options for importing Max objects when updating
+    or switching assets.
+    """
+    def __init__(self, project_templates):
+        super(SubstanceProjectConfigurationWindow, self).__init__()
+        self.setWindowFlags(self.windowFlags() | QtCore.Qt.FramelessWindowHint)
+
+        self.template_name = None
+        self.project_templates = project_templates
+
+        self.widgets = {
+            "label": QtWidgets.QLabel("Project Configuration"),
+            "template_options": QtWidgets.QComboBox(),
+            "buttons": QtWidgets.QWidget(),
+            "okButton": QtWidgets.QPushButton("Ok"),
+        }
+        for template in project_templates:
+            self.widgets["template_options"].addItem(template)
+        # Build buttons.
+        layout = QtWidgets.QHBoxLayout(self.widgets["buttons"])
+        layout.addWidget(self.widgets["template_options"])
+        layout.addWidget(self.widgets["okButton"])
+        # Build layout.
+        layout = QtWidgets.QVBoxLayout(self)
+        layout.addWidget(self.widgets["label"])
+        layout.addWidget(self.widgets["buttons"])
+
+        self.widgets["okButton"].pressed.connect(self.on_ok_pressed)
+
```

Comment:
I would build it so that the "presets" are just that - presets, but in the end each preset still sets the individual attributes in the dialog so that the artist can use a preset as a basis to quickly select a preset - but still have full control and see what the individual settings are.

Currently this just lists the presets and offers no visual difference to the "options" dialog you had before from loader options?

My point was to provide the user ALL the options they want - including picking resolution, UDIM or not, etc. so that to the artist it still feels like how they would configure substance painter.

However, since you mentioned someone wanted presets - then we can expose 'presets' from settings to the dialog to quickly toggle those attributes to "preset values" from the settings so you have both, but still allow customization by the artist. No?

Reply:
I have discussed with @LiborBatek about this.  The reason why there is template setting to make the workflow more streamlined. It's necessary for the manager/TD to align the project settings so that it is more unified when the texture exported from substance and used it in other DCCs

Reply:
> It's necessary for the manager/TD to align the project settings so that it is more unified when the texture exported from substance and used it in other DCCs

I guess fine with me - then maybe later I might expose a setting that allows to enable customization by the artist instead of picking from presets.